scripts/check_type_hints.py

```
@@ -194,13 +225,98 @@ def add_future_import(file_path: Path) -> bool:
+
+        insert_index = _find_insert_index(lines, content)
+        _insert_import_line(lines, insert_index)
+
         file_path.write_text("".join(lines), encoding="utf-8")
         return True
     except Exception as e:
         print(f"Error adding import to {file_path}: {e}", file=sys.stderr)
         return False
 
 
+def _find_python_files(src_dir: Path, include_tests: bool) -> list[Path]:
```

Ruff (0.14.7)

Boolean-typed positional argument in function definition

(FBT001)